### PR TITLE
[docs] PR template to include commit message

### PR DESCRIPTION
**Pull request by asraa**
Commit message: Add a "Commit message" field to PR templates that will be used when merging commits. Use "Additional description" for extra details.
Description: Updated the contributing guidelines/maintainer notes to make sure PR messages/descriptions are updated as they're reviewed, and that the commit message can be used as the default commit message when merging.  
Risk Level: Low
Fixes: https://github.com/envoyproxy/envoy/issues/10891

Signed-off-by: Asra Ali <asraa@google.com>

### CODE REVIEW

File: PULL_REQUESTS.md [hunk 1]
```diff
@@ -12,13 +12,21 @@ explaining the overall change. Both the component and the explanation must be lo
...
-how the feature is enabled etc...
+The commit message field should include an explanation of what this PR
+does. This will be used as the final commit message that maintainers will use to
+populate the commit message when merging. If this PR causes a change in behavior
+it should document the behavior before and after. If fixing a bug, please
+describe what the original issue is and how the change resolves it. If it is
+configuration controlled, it should note how the feature is enabled etc...
+
+
+### <a name="desc"></a>Detailed Description
```
Comment (asraa) [inline PULL_REQUESTS.md:25]: Maybe this should be "Additional Description" to indicate that the meat of their description should belong in the commit message?

**Comment by asraa** [comment]
cc @mpuncel

File: PULL_REQUESTS.md (hunk 1 above)
Comment (mattklein123) [reply PULL_REQUESTS.md:25]: Yeah makes sense to me.

Comment (asraa) [reply PULL_REQUESTS.md]: all set!

**Pull request merged**